release-notes/swan-lake-2201.1.0-release-note.md

```diff
@@ -30,9 +30,386 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+    io:println(decimal:quantize(123.123, 1.00)); // 123.12
+    io:println(decimal:quantize(123.123, 1.000)); // 123.123
+}
+```
+If the length of the coefficient after the quantize operation would be greater than precision, then an `InvalidOperation` is signaled.
+
+```
+public function main() {
+	decimal _ = decimal:quantize(123.1233, 1E-36); // results in an error
+}
+```
+
+###### New `lang.float:toFixedString()` and `lang.float:toExpString()` functions
+
+Two new functions namely `toFixedString()` and `toExpString()` have been introduced to get the string representation of a `float` value in fixed-point notation and scientific notation respectively. Both the functions facilitate you to specify the number of digits required to be followed by the decimal point.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    string a = float:toFixedString(5.7, 16); 
+    io:println(a); // 5.7000000000000002
+    string b = float:toFixedString(5.7, 2); 
+    io:println(b); // 5.70
+    
+    float f1 = -45362.12334;
+    string c = float:toExpString(f1, 16); 
+    io:println(c); // -4.5362123339999998e+4
+    string d = float:toExpString(f1, 2); 
+    io:println(d); // -4.54e+4
+}
+```
+
+###### New `lang.string:padStart()`, `lang.string:padEnd()`, and `lang.string:padZero()` functions
+
+`padStart()`, `padEnd()`, and `padZero()` functions have been introduced to add padding in strings. `padStart` adds padding to the start of a string. `padEnd` adds padding to the end of a string. `padZero` pads a string with zeros.
+
+```ballerina
+public function main() {
+    io:println("abc".padStart(5, "#")); // "##abc"
+    io:println("abc".padStart(5)); // "  abc"
+
+    io:println("abc".padEnd(5, "#")); // "abc##"
+    io:println("abc".padEnd(5)); // "abc  "
+
+    io:println("123".padZero(5)); // "00123"
+    io:println("123".padZero(5, "#")); // "##123"
+}
+```
+
+#### Improvements
+
+##### Disallow inferring array sizes from contexts that are not permitted
+
+Inferring array length from the context is restricted to support only with list constructors. Further, only the first dimension can be inferred from the context in multidimensional arrays.
+
+```ballerina
+int[*] x1 = [1, 2]; // Supported.
+
+int[2] y = [1, 2];
+int[*] x2 = y; // Not supported. Required a list constructor to infer            the array length.
+
+int[*][2] x3 = [[1, 2], [1, 2]]; // Supported.
+int[*][*] x4 = [[1, 2], [1, 2]]; // Not supported. Only the first dimension can be inferred from the context.
+```
+
+##### Add annotation attachments to the BIR
+
+
+
+##### Revamp round langlib function for float
+
+The method signature is changed to have an extra argument `fractionDigits`, where the user can choose the number of fraction digits of the rounded result. When `fractionDigits` is zero, the method rounds to an integer.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    io:println(555.545.round(2)); // 555.54
```

Comment:
@KavinduZoysa  @KRVPerera  Shall we change this also like the decimal example?

Reply:
I am sending a PR to change both 
https://github.com/gimantha/ballerina-release/pull/2

@KavinduZoysa